packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -2,17 +2,153 @@ import 'clay-button';
-import Soy from 'metal-soy';
+import {Align} from 'metal-position';
+import {core, object} from 'metal';
 import {Config} from 'metal-state';
-
+import Component from 'metal-component';
+import dom from 'metal-dom';
+import {EventHandler} from 'metal-events';
+import Soy from 'metal-soy';
 import itemsValidator from './items_validator';
 import templates from './ClayDropdownBase.soy.js';
 
 /**
  * Implementation of the base for Metal Clay Dropdown.
  */
-class ClayDropdownBase extends Dropdown {
+class ClayDropdownBase extends Component {
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		super.attached();
+		this.eventHandler_.add(
+			dom.on(document, 'click', this.handleDocClick_.bind(this))
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.eventHandler_ = new EventHandler();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	detached() {
+		super.attached();
+		this.eventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Closes the dropdown.
+	 * @protected
+	 */
+	close_() {
+		this.expanded = false;
+	}
+
+	/**
+	 * Handles document click in order to hide menu.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleDocClick_(event) {
+		if (this.element.contains(event.target)) {
+			return;
+		}
+		this.close_();
+	}
+
+	/**
+	 * Toggles the dropdown, closing it when open or opening it when closed.
+	 */
+	toggle() {
+		this.expanded = !this.expanded;
+	}
+
+	/**
+	 * The setter function for the `classMap_` staet.
+	 * @param {Object} value
+	 * @return {!object}
+	 * @protected
+	 */
+	setterClassMapFn_(value) {
+		return object.mixin(this.valueClassMapFn_(), value);
+	}
+
+	/**
+	 * The setter function for the `position` state. Converts the supported
+	 * string positions into the appropriate `Align` position constants.
+	 * @param {string|number} value
+	 * @return {number}
+	 * @protected
+	 */
+	setterPositionFn_(value) {
+		if (core.isNumber(value)) {
+			return value;
+		}
+		return value.toLowerCase() === 'up' ? Align.TopLeft : Align.BottomLeft;
+	}
+
+	/**
+	 * Synchronization logic for `expanded` state.
+	 * @param {boolean} expanded
+	 */
+	syncExpanded(expanded) {
+		if (expanded && this.alignElementSelector) {
+			// eslint-disable-next-line
+			let alignElement = this.element.querySelector(this.alignElementSelector);
+			if (alignElement) {
+				let bodyElement = this.element.querySelector('.dropdown-menu');
+				this.alignedPosition_ = Align.align(
+					bodyElement,
+					alignElement,
+					this.position_
+				);
+			}
+		}
+	}
+
+	/**
+	 * Validator for the `position` state.
+	 * @param {string|number} position
+	 * @return {boolean}
+	 * @protected
+	 */
+	validatePosition_(position) {
+		if (Align.isValidPosition(position)) {
+			return true;
+		}
+
+		switch (position.toLowerCase()) {
+		case 'up':
+		case 'down':
+			return true;
+		default:
+			return false;
+		}
+	}
+
+	/**
+	 * Gets the default value for the `classMap_` state.
+	 * @return {!Object}
+	 * @protected
+	 */
+	valueClassMapFn_() {
```

Comment:
classMap was used in metal-dropdown soy template. Since we're not using this param in our templates we can get rid of this method.